service/src/NVMLIOGroup.cpp

```diff
@@ -236,26 +275,58 @@ namespace geopm
             }
             sv.second.controls = result;
         }
-        register_control_alias("GPU_POWER_LIMIT_CONTROL", M_NAME_PREFIX + "GPU_POWER_LIMIT_CONTROL");
-        register_signal_alias("GPU_POWER_LIMIT_CONTROL", M_NAME_PREFIX + "GPU_POWER_LIMIT_CONTROL");
-        register_control_alias("GPU_CORE_FREQUENCY_CONTROL", M_NAME_PREFIX + "GPU_CORE_FREQUENCY_CONTROL");
-        register_signal_alias("GPU_CORE_FREQUENCY_CONTROL", M_NAME_PREFIX + "GPU_CORE_FREQUENCY_CONTROL");
 
-        for (int domain_idx = 0; domain_idx < m_platform_topo.num_domain(GEOPM_DOMAIN_GPU); ++domain_idx) {
-            std::vector<unsigned int> supported_frequency = m_nvml_device_pool.frequency_supported_sm(domain_idx);
-            if (supported_frequency.size() == 0) {
-                // TODO: Long term this should simply hide the FREQUENCY_MIN and FREQUENCY_MAX signals,
-                //       not prevent loading of the IO Group
+        // Cache the initial settings
+        save_control();
 
-                throw Exception("NVMLIOGroup::" + std::string(__func__) +
-                                ": No supported frequencies found for GPU " + std::to_string(domain_idx),
-                                GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        std::vector <std::string> unsupported_control_names;
+        for (auto &sv : m_control_available) {
+            try {
+                double init_setting = NAN;
+                if(is_valid_signal(sv.first) &&
+                   signal_domain_type(sv.first) == control_domain_type(sv.first)) {
+                    for (int domain_idx = 0;
+                         domain_idx < m_platform_topo.num_domain(control_domain_type(sv.first));
+                         ++domain_idx) {
+                        init_setting = read_signal(sv.first, control_domain_type(sv.first), domain_idx);
+
+                        if(std::isnan(init_setting)) {
+                            // Specialized handling for signals that may be NAN
+                            if (sv.first == M_NAME_PREFIX + "GPU_CORE_FREQUENCY_RESET_CONTROL") {
+                                init_setting = 0;
+                            }
+                        }
+                        else if (init_setting == 0 &&
```

Comment:
It's unclear to me why the default GPU_CORE_FREQUENCY_CONTROL value is 0 instead of NAN (defined in initializer list), but either way this is handled as a special case by using the max frequency for the write/read check then calling restore_control.